taipy/core/scenario/scenario.py

```diff
@@ -175,11 +176,40 @@ def __hash__(self):
     def __eq__(self, other):
         return isinstance(other, Scenario) and self.id == other.id
 
-    def __getattr__(self, attribute_name):
-        protected_attribute_name = _validate_id(attribute_name)
-        if protected_attribute_name in self._properties:
-            return _tpl._replace_templates(self._properties[protected_attribute_name])
-
+    def __setattr__(self, name: str, value: Any) -> None:
+        entity_taipy_attributes = [
+            "_config_id",
+            "id",
+            "_tasks",
+            "_additional_data_nodes",
+            "_creation_date",
+            "_cycle",
+            "_primary_scenario",
+            "_tags",
+            "_properties",
+            "_sequences",
+            "_version",
+            "_submittable_id",
+            "_subscribers",  # from Submittable class
+            "_in_context_attributes_changed_collector",
+            "_is_in_context"  # from _Entity class
+            "name",  # from def name setter
```

Comment:
What if we use `Scenario.__dict__.keys()` here?
Will that also cause a recursion error?

Reply:
well Scenario.__dict__ only contains pre-defined functions or attributes outside of __init__, while we need the ones within __init__ also and they are not set/available until __setattr__ is called

Reply:
well Scenario.__dict__ only contains pre-defined functions or attributes outside of __init__, while we need the ones within __init__ also and they are not set/available until __setattr__ is called

Reply:
right, this is tough